ion-schema/src/result.rs

```diff
@@ -40,6 +40,26 @@ pub enum IonSchemaError {
     },
 }
 
+// io::Error does not implement PartialEq, which precludes us from simply deriving an implementation.
+impl PartialEq for IonSchemaError {
```

Comment:
I believe that it is reflexive. If if you implement or derive `Eq` you get `PartialEq` for free.

Reply:
Yes, I have added `Eq` implementation of empty block as a marker trait.

Reply:
I wasn't saying that we must mark it as `Eq`. I'm legitimately asking if there's any reasons for or against it. 

It doesn't add functionality, but does it enable any specific use cases? Or, are we tying our hands for future changes if we add `Eq`?

Reply:
I don't see any reasons against adding an `Eq` implementation which is the reason why added it. Which is why I have added `Eq` implementation along with `PartialEq` where the `Eq` implementation is a marker trait and has no methods because `PartialEq` is sufficient for reflexive equality. 
`Eq` can be useful for `Hash` implementation later if someone wants to use the error as key but other than that I dont know of any specific use case.
Does this answer your questions?

Reply:
I think we should hold off on adding an `Eq` implementation until we have a real use case for it. `Eq` puts some modest constraints on the evolution of that type that we shouldn't take on until someone requests it.